src/frequenz/sdk/timeseries/_formula_engine/_formula_engine.py

```diff
@@ -55,15 +55,15 @@
 }
 
 
-class FormulaEvaluator:
+class FormulaEvaluator(Generic[GenericQuantity]):
     """A post-fix formula evaluator that operates on `Sample` receivers."""
 
     def __init__(
         self,
         name: str,
         steps: List[FormulaStep],
-        metric_fetchers: Dict[str, MetricFetcher],
-        output_type: Type[Quantity],
+        metric_fetchers: Dict[str, MetricFetcher[GenericQuantity]],
+        output_type: Type[GenericQuantity],
```

Comment:
Can't you just use `GenericQuantity` when you need `output_type`? Do you really need to store it?

Reply:
We need to store it in places where objects of that type need to be created.  For example,  a FormulaEngine might take dimensionless values directly from the resampler and return Power values.  Because the interpreter doesn't understand typehints,  we have to give it an actual class to use.